src/main/clojure/clara/rules.cljc

```diff
@@ -311,14 +311,17 @@
   (defmacro mk-session
      "Creates a new session using the given rule sources. The resulting session
       is immutable, and can be used with insert, retract, fire-rules, and query functions.
-
-      If no sources are provided, it will attempt to load rules from the caller's namespace,
-      which is determined by reading Clojure's *ns* var.
+      
+      `sources-and-args` can start with 0+ sources, each being either a namespace symbol or
```

Comment:
I think this is an improvement to the doc string overall.

I do think it's a bit more strict in the doc string than the actual fn is though.

The concept of a "rule source" is protocol-based in clara. The caller could give other data types that represent rule sources.

There is a clara.rules.compiler/IRuleSource protocol that has a few built-in extensions to it for things like clojure.lang.Symbol [as can be seen here](https://github.com/cerner/clara-rules/blob/0.21.2/src/main/clojure/clara/rules.cljc#L274). People can extend it to other things, eg. clojure.lang.Var types if they wanted to (I've done this before). If what is given does not satisfy this protocol, then it is assumed to indeed be a sequence of rule/query (aka "production") data structures. This logic [can be observed here](https://github.com/cerner/clara-rules/blob/0.21.2/src/main/clojure/clara/rules/compiler.clj#L2097).